src/app/security/maturity-model/components/MobileTableOfContents.tsx

```diff
@@ -0,0 +1,56 @@
+'use client'
+
+import { Listbox as HeadlessUIListbox } from '@headlessui/react'
+import { ArrowsDownUp, CaretDown } from '@phosphor-icons/react/dist/ssr'
+
+import { Icon } from '@/components/Icon'
+import { ListboxButton } from '@/components/ListboxButton'
+import { ListboxOption } from '@/components/ListboxOption'
+import { ListboxOptions } from '@/components/ListboxOptions'
+
+import { coreFunctionsData } from '../data/coreFunctionsData'
+import { scrollToSection } from '../utils/scrollToSection'
+import { useUrlHash } from '../utils/useUrlHash'
+
+const options = coreFunctionsData.map(({ slug, title }) => ({
+  id: slug,
+  name: title,
+}))
+
+export function MobileTableOfContents() {
+  const { updateHash, isSectionActive, getHashFromSlug } = useUrlHash()
+
+  const selectedOption = options.find((option) => isSectionActive(option.id))
```

Comment:
Your message is mostly clear, but it can be refined for better flow and clarity. Here’s a revised version:

This isn’t working as expected. Please refer to the console errors and browser behavior.

An easy fix would be:

```
const defaultOption = options[0]
const selectedOption = options.find((option) => isSectionActive(option.id)) || defaultOption
```

However, as mentioned in the other PR, it would be ideal if we didn’t have to do this workaround.

We can try using:

`const defaultOption = coreFunctionsData.find((item) => item.order === 1)!`

…with the non-null assertion (!) to enforce a default, but it seems like coreFunctionsBaseData isn’t enforcing the order type. Could we ensure that the order type is properly handled in coreFunctionsBaseData?

Reply:
If we want `.find()` to always return a non-null value in places where we know the value exists for sure, we can create this helper function:

```ts
function findOrFail<T>(array: T[], fn: (item: T) => boolean): T {
  const found = array.find(fn);
  if (!found) {
    throw new Error('Item not found');
  }
  return found;
}
```

If someday the value doesn't exist anymore, the build will crash :)

Reply:
>Regarding having a default, coreFunctionsBaseData is properly typed; 

When I changed `order` to `order: 0` in `/src/app/security/maturity-model/content/leadership-and-organizational-principles.md` - Nothing yelled at me.

Maybe `findOrFail` is in this case a good solution. It seems important to me that the first chapter would be the default? And not the second, etc. Though, I don't know if this is ever an issue, since it seems to already populate correctly by the time the user scrolls to the beginning?

Reply:
I added a zod validation and a sort function so that the first chapter will always be first! If you change the order to zero -> fails